### PR TITLE
[AIR][Docs] Remove `BatchPredictor` from examples

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

#36947 deprecated `BatchPredictor`. This PR removes or updates references to `BatchPredictor` in the AIR examples.

The following examples already use updated APIs:
* Stable Diffusion Batch Prediction with Ray AIR
* GPT-J-6B Batch Prediction with Ray AIR (LLM)

The following examples have been updated to use updated APIs:
* Training a model with distributed XGBoost
* Training a model with distributed LightGBM

I've removed batch prediction sections from the other examples, and, where appropriate, linked to the batch inference user guide.

## Related issue number

Towards https://github.com/ray-project/ray/issues/37035

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by pcmoritz** [comment]
Thanks a lot for doing this ❤️ 

Do you think all the removed material is already covered in the Ray Data batch inference docs? How about keeping some of the examples (e.g. the starter examples), just convert them to `map_batches` instead and also link to the batch inference docs ("if you want to learn more about batch inference, look here: ..."). This way the blow would be softened a bunch and the end-to-end nature of Ray AIR would be preserved.

Some of the stuff can probably be removed yes :)

**Comment by pcmoritz** [comment]
Also curious about @richardliaw 's opinion here

**Review by ericl — APPROVED** [review]

File: doc/source/ray-air/examples/convert_existing_pytorch_code_to_ray_air.ipynb [hunk 1]
```diff
@@ -18,13 +18,13 @@
     "- Automatic checkpointing/fault tolerance and result tracking\n",
```
Comment (richardliaw) [inline doc/source/ray-air/examples/convert_existing_pytorch_code_to_ray_air.ipynb:18]: Comment from @pcmoritz - this seems worth updating

File: doc/source/ray-air/examples/opt_deepspeed_batch_inference.ipynb [hunk 2]
```diff
@@ -1,911 +0,0 @@
-{
```
Comment (richardliaw) [inline doc/source/ray-air/examples/opt_deepspeed_batch_inference.ipynb:1]: maybe this one as well?

File: doc/source/ray-air/examples/convert_existing_pytorch_code_to_ray_air.ipynb (hunk 1 above)
Comment (richardliaw) [reply doc/source/ray-air/examples/convert_existing_pytorch_code_to_ray_air.ipynb:18]: We should just link this one to batchinference.io

**Pull request merged**